### PR TITLE
added metadata to legends_item_visibility.py

### PR DESCRIPTION
**Pull request by wenebob**
This is my new pull request. Thanks
All pull requests must have an associated issue in the issue tracker. If there
isn't one, please go open an issue describing the defect, deficiency or desired
feature. You can read more about our issue and PR processes in the
[wiki](https://github.com/bokeh/bokeh/wiki/BEP-1:-Issues-and-PRs-management).

- [ ] issues: addresses #11765 
- [ ] tests added / passed
- [ ] release document entry (if new feature or API change)

### CODE REVIEW

File: examples/basic/annotations/legends_item_visibility.py [hunk 1]
```diff
@@ -1,3 +1,11 @@
+''' Marker and line plots that demonstrate manual control of legend visibility of individual items
+
+.. bokeh-example-metadata::
+    :apis: bokeh.plotting.figure.circle, bokeh.plotting.figure.line
+    :refs: :ref:`ug_basic_annotations_hiding-legend-items`
```
Comment (bryevdv) [inline examples/basic/annotations/legends_item_visibility.py]: this reference anchor does not exist in the sphinx docs source under `docs/bokeh`

There is a general `ug_basic_annotations` that could be used, @tcmetzger do you know if there are any better references that could be used?

Comment (wenebob) [reply examples/basic/annotations/legends_item_visibility.py]: Do you mean `hiding-legend-items` @bryevdv? 
If yes, then I took it from this [link](https://docs.bokeh.org/en/latest/docs/user_guide/basic/annotations.html#hiding-legend-items)
Please let me know how to correct it.
Thanks

Comment (bryevdv) [reply examples/basic/annotations/legends_item_visibility.py]: Hi @Faith-Nchifor The Bokeh docs are built using a tool called Sphinx which processes ReStructured Text (ReST) files to build the final HTML. The valid references are actually defined in the ReST files:

https://raw.githubusercontent.com/bokeh/bokeh/branch-3.2/docs/bokeh/source/docs/user_guide/basic/annotations.rst

(I had to link to a raw URL since GitHUb tries to render ReST files and that hides what I want to show below)

So in that file, e.g.
```
.. _ug_basic_annotations_titles:
```
would make `ug_basic_annotations_titles` a valid reference. We've kind of glossed over this bit in the issue since there is already a lot going on and we can usually just help by providing the necessary references. But you asked for details :D 

Long story short: I would just change to 
```
:refs: :ref:`ug_basic_annotations
``` 
for now, until/unless @tcmetzger can offer any other appropriate suggestions.

Comment (wenebob) [reply examples/basic/annotations/legends_item_visibility.py]: @bryevdv okay sir. It's been corrected.
Thank you for sharing that link. I have seen what needs to be adjusted. It has to be in this format: `ug_basic_annotations_legends_item_visibility`. It's found in this file gist you shared. If it's okay with you I will go ahead and make the changes. Otherwise I will wait for @tcmetzger to confirm it.

File: examples/basic/annotations/legends_item_visibility.py [hunk 2]
```diff
@@ -1,3 +1,11 @@
+''' Marker and line plots that demonstrate manual control of legend visibility of individual items
+
+.. bokeh-example-metadata::
+    :apis: bokeh.plotting.figure.circle, bokeh.plotting.figure.line
+    :refs: :ref:`ug_basic_annotations`
```
Comment (tcmetzger) [inline examples/basic/annotations/legends_item_visibility.py]: I think we have a section specifically about legend item visibility (which is actually using this example), so you could do this:

```suggestion
    :refs: :ref:`ug_basic_annotations` > :ref:`ug_basic_annotations_legends_item_visibility`
```

File: examples/basic/annotations/legends_item_visibility.py [hunk 3]
```diff
@@ -1,3 +1,11 @@
+''' Marker and line plots that demonstrate manual control of legend visibility of individual items
+
+.. bokeh-example-metadata::
+    :apis: bokeh.plotting.figure.circle, bokeh.plotting.figure.line
```
Comment (tcmetzger) [inline examples/basic/annotations/legends_item_visibility.py]: This uses the LegendItem object, at least behind the scenes, so we could add that here as well:

```suggestion
    :apis: bokeh.models.LegendItem, bokeh.plotting.figure.circle, bokeh.plotting.figure.line
```

**Comment by wenebob** [comment]
These changes have been made as suggested. Thank you @tcmetzger

File: examples/basic/annotations/legends_item_visibility.py [hunk 4]
```diff
@@ -1,3 +1,11 @@
+''' Marker and line plots that demonstrate manual control of legend visibility of individual items
+
+.. bokeh-example-metadata::
+    :apis: bokeh.models.LegendItem, bokeh.plotting.figure.circle, bokeh.plotting.figure.line
+    :refs: :ref:`ug_basic_annotations` > :ref:`ug_basic_annotations_legends_item_visibility`
```
Comment (bryevdv) [inline examples/basic/annotations/legends_item_visibility.py]: Last comment: we stopped doing the "breadcrumb" links with `>` because they are just too much of a burden to maintain  manually. These just need the direct reference to the section, which is simpler:
```suggestion
    :refs: :ref:`ug_basic_annotations_legends_item_visibility`
```

**Comment by wenebob** [comment]
I have made the change @bryevdv 
Thank you

**Review by bryevdv — APPROVED** [review]

**Comment by bryevdv** [comment]
Windows CI failures are unrelated. Thanks @Faith-Nchifor !
